src/zeroconf/_dns.py

```diff
@@ -222,12 +225,7 @@ def is_recent(self, now: _float) -> bool:
         """Returns true if the record more than one quarter of its TTL remaining."""
         return self.created + (_RECENT_TIME_MS * self.ttl) > now
 
-    def reset_ttl(self, other) -> None:  # type: ignore[no-untyped-def]
-        """Sets this record's TTL and created time to that of
-        another record."""
-        self.set_created_ttl(other.created, other.ttl)
-
-    def set_created_ttl(self, created: _float, ttl: Union[float, int]) -> None:
+    def _set_created_ttl(self, created: _float, ttl: Union[float, int]) -> None:
         """Set the created and ttl of a record."""
```

Comment:
This isn't great that we mutate existing records after they are in the cache and downstream has them. We really should copy them and put the new record in the cache but we don't have a copy implementation so its the best we can do right now